src/librustc_trans/back/write.rs

```diff
@@ -557,9 +561,15 @@ unsafe fn optimize_and_codegen(cgcx: &CodegenContext,
 
         if config.emit_obj {
             let path = output_names.with_extension(&format!("{}.o", name_extra));
-            with_codegen(tm, llmod, config.no_builtins, |cpm| {
-                write_output_file(cgcx.handler, tm, cpm, llmod, &path, llvm::ObjectFileType);
-            });
+            if !cgcx.is_like_pnacl {
+                with_codegen(tm, llmod, config.no_builtins, |cpm| {
```

Comment:
This may just be me being ignorant of the LLVM fork for pnacl, but if you're already forking LLVM couldn't support for this be added? In terms of interoperating with other compilers it seems like it'd be best to put logic into the pnacl target rather than every compiler using LLVM?